engines/data_export/lib/data_export/engine.rb

```diff
@@ -27,16 +27,12 @@ def export_rmt_data
       end
 
       Api::Connect::V3::Systems::SystemsController.class_eval do
-        after_action :export_rmt_data, only: %i[update], if: -> { response.successful? && !@system.byos? }
+        after_action :export_rmt_data, only: %i[update], if: -> { response.successful? && !@system.byos? && !@system.products.empty? }
```

Comment:
You could use the `Object#present?` method here instead of `!@system.products.empty?`
But it's authors choice.
```suggestion
        after_action :export_rmt_data, only: %i[update], if: -> { response.successful? && !@system.byos? && @system.products.present? }
```